### ISSUE TITLE
[Bug]: Delete by id successfully, the id can not be queried but can still be searched

### ISSUE THREAD

**Issue by 530513750** [milvus-io/milvus#16993]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.0.2
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):2.0.2
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: eight core / 16G
- GPU: None
- Others:
```

### Current Behavior

我用docker-compose在单机上搭建的2.0.2分布式版本，创建和搜索的时候都加上了一致性等级的参数consistency_level="Strong" 还是有那个问题，按条件delete(expr="id in [2]")删除后，query(expr="id in [2]")不到数据，但是用向量特征可以search()到删除的数据id。下面是我测试的代码，可以直接执行来复现
from pymilvus import (
    connections,
    utility,
    FieldSchema,
    CollectionSchema,
    DataType,
    Collection,
)
import random
def extractor():
    vects=[[random.random() for _ in range(128)] for _ in range(10)] #
    return vects

def add(collection_name,imgFeatureVector,id,nlist=256):
    vector_name="vects"
    fields = [
        FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=False),
        FieldSchema(name=vector_name, dtype=DataType.FLOAT_VECTOR, dim=128)
    ]
    schema = CollectionSchema(fields, "book content")
    milvus = Collection(collection_name, schema, consistency_level="Strong") #

    #这里插入了相同id的len(imgFeatureVector)条记录
    data = [
        [id for i in range(len(imgFeatureVector))],
        imgFeatureVector
    ]
    mr = milvus.insert(data)
    print(mr)
    if not milvus.has_index():
        index_params = {
            "metric_type": "L2",
            "index_type": "IVF_SQ8",
            "params": {"nlist": nlist}
        }
        print("Creating index:",collection_name)
        milvus.create_index(field_name=vector_name, index_params=index_params)
    return True

def search(collection_name,query_vectors,nprobe=2,partition_tags=None):
    search_param = {
        "metric_type": "L2",
        "params": {"nprobe": nprobe}
    }
    milvus = Collection(name=collection_name, consistency_level="Strong") #
    milvus.load()
    print("Searching ... ",len(query_vectors))
    param = {
        'data': query_vectors,
        'anns_field':'vects',
        'param': search_param,
        'limit': 1,
        'partition_names':partition_tags
    }
    results = milvus.search(**param)
    print("search results:",results)

def deleteId(collection_name,id):#删除一条记录
    milvus = Collection(collection_name, consistency_level="Strong")
    r = milvus.delete(expr="id in ["+str(id)+"]")
    print("deleteId results:",r)

def getId(collection_name,id):#获取一条记录向量信息 目前重复的主键id只能获取到一条
    milvus = Collection(collection_name, consistency_level="Strong")
    r = milvus.query(expr="id in [" + str(id) + "]", output_fields=["vects"])
    print("getId results:",r)

if __name__ == '__main__':
    connections.connect("default", host="localhost", port="19530")
    collection_name = 'test_name'
    #第一步先创建集合并生成索引
    for id in range(1, 5):
        fea = extractor()
        add(collection_name, fea, id)

    #随机取部分特征数据先执行一下搜索可以搜索到所有的数据
    fea = extractor()
    search(collection_name,fea)

    #测试删除
    getId(collection_name, 2)  # 打印出来有对应的记录
    deleteId(collection_name,2) #删除id=2的记录
    getId(collection_name,2) #打印出来确实没有对应的记录了
    search(collection_name, fea) #问题出在这里，为什么删除的数据但是这里仍然可以搜索到，希望大神能解答

附上我执行上面代码打印的一个结果
(insert count: 10, delete count: 0, upsert count: 0, timestamp: 433193008136716293)
Creating index: test_name
(insert count: 10, delete count: 0, upsert count: 0, timestamp: 433193008935993350)
(insert count: 10, delete count: 0, upsert count: 0, timestamp: 433193008949624838)
(insert count: 10, delete count: 0, upsert count: 0, timestamp: 433193008949624846)
Searching ...  10
search results: ["['(distance: 18.288646697998047, id: 4)']", "['(distance: 17.622722625732422, id: 4)']", "['(distance: 17.082468032836914, id: 4)']", "['(distance: 15.920527458190918, id: 4)']", "['(distance: 15.110637664794922, id: 4)']", "['(distance: 15.388628005981445, id: 4)']", "['(distance: 17.998126983642578, id: 2)']", "['(distance: 17.456764221191406, id: 2)']", "['(distance: 17.85945701599121, id: 2)']", "['(distance: 18.325977325439453, id: 2)']"]
getId results: [{'id': 2, 'vects': [0.068687, 0.824321, 0.32938, 0.224203, 0.607252, 0.670809, 0.953021, 0.838718, 0.080241, 0.256887, 0.376128, 0.739589, 0.096422, 0.750901, 0.367866, 0.60797, 0.533248, 0.864351, 0.915973, 0.734446, 0.029596, 0.6837, 0.742087, 0.835008, 0.796056, 0.81758, 0.85339, 0.465357, 0.79914, 0.873137, 0.220855, 0.332924, 0.281748, 0.796896, 0.715476, 0.303109, 0.213279, 0.393482, 0.830795, 0.437887, 0.156069, 0.126897, 0.490367, 0.331049, 0.960818, 0.539621, 0.076401, 0.3409, 0.589615, 0.413125, 0.771736, 0.421611, 0.395455, 0.472049, 0.640238, 0.053566, 0.711115, 0.787823, 0.881577, 0.813229, 0.661634, 0.663761, 0.924615, 0.340838, 0.541913, 0.251503, 0.922315, 0.765425, 0.115499, 0.168405, 0.160131, 0.383442, 0.372767, 0.8945, 0.962986, 0.491972, 0.922999, 0.150437, 0.045667, 0.553842, 0.421755, 0.981189, 0.9207, 0.530157, 0.833767, 0.349062, 0.924295, 0.521678, 0.615996, 0.385622, 0.673163, 0.340106, 0.099098, 0.953671, 0.564427, 0.980742, 0.928303, 0.422721, 0.810244, 0.999839, 0.862782, 0.006507, 0.325584, 0.730744, 0.271744, 0.944179, 0.038029, 0.09106, 0.460184, 0.119262, 0.810694, 0.891065, 0.589844, 0.419055, 0.522863, 0.463735, 0.811044, 0.664067, 0.341331, 0.728953, 0.713194, 0.823001, 0.017049, 0.554018, 0.236049, 0.43507, 0.952209, 0.90193]}]
deleteId results: (insert count: 0, delete count: 1, upsert count: 0, timestamp: 433193009342578694)
getId results: []
Searching ...  10
search results: ["['(distance: 18.288646697998047, id: 4)']", "['(distance: 17.622722625732422, id: 4)']", "['(distance: 17.082468032836914, id: 4)']", "['(distance: 15.920527458190918, id: 4)']", "['(distance: 15.110637664794922, id: 4)']", "['(distance: 15.388628005981445, id: 4)']", "['(distance: 17.998126983642578, id: 2)']", "['(distance: 17.456764221191406, id: 2)']", "['(distance: 18.038719177246094, id: 2)']", "['(distance: 18.325977325439453, id: 2)']"]

**Comment by yanliang567** [comment]
@xige-16 I think you are trying to refactor on this feature.
/assign @xige-16 
/unassign

**Comment by xige-16** [comment]
I'm looking into this isssue.

**Comment by xige-16** [comment]
This search result is very strange, milvus will deduplicate according to the pk in the search result, but this result still has duplicate pk

**Comment by xige-16** [comment]
> This search result is very strange, milvus will deduplicate according to the pk in the search result, but this result still has duplicate pk

My mistake, the input of search is 10 vectors, topk=1, so the repeated pk is not in the same nq.

**Comment by xige-16** [comment]
This is a bug of delete, it has been fixed on master, the problem is very complicated. . .　@530513750

**Comment by xige-16** [comment]
releated pr https://github.com/milvus-io/milvus/pull/16863

**Comment by xige-16** [comment]
> This is a bug of delete, it has been fixed on master, the problem is very complicated. . .　@530513750

The problem is that if a duplicate pk is inserted, for example, insert pk1 first, then insert pk1, and then delete pk1, then only the second inserted pk1 is deleted, and the first inserted pk1 can still be queried. This bug has been fixed.

**Comment by 530513750** [comment]
> 

Thanks for your reply, indeed, my data have duplicate pk. I want know how can I delete all data "id=2" in the collection. Another problem, why I delete one of the duplicate pk record, but query() results is "[]"

**Comment by 530513750** [comment]
> > This is a bug of delete, it has been fixed on master, the problem is very complicated. . .　@530513750
> 
> The problem is that if a duplicate pk is inserted, for example, insert pk1 first, then insert pk1, and then delete pk1, then only the second inserted pk1 is deleted, and the first inserted pk1 can still be queried. This bug has been fixed.

Thanks for your reply, indeed, my data have duplicate pk. I want know how can I delete all data "id=2" in the collection. Another problem, why I delete one of the duplicate pk record, but query() results is "[]"

**Comment by yanliang567** [comment]
I believe this was fixed in master build, could you please help to verify it on a master build or Milvus 2.1 release(will be available next week) @530513750 

/assign @530513750 
/unassign
